esphome/components/ethernet/__init__.py

```diff
@@ -107,6 +114,18 @@ def _validate(config):
         else:
             use_address = CORE.name + config[CONF_DOMAIN]
         config[CONF_USE_ADDRESS] = use_address
+    if config.get(CONF_TYPE) == "LAN867X":
+        # Check if the ESP-IDF framework is used
+        if not CORE.using_esp_idf:
+            raise cv.Invalid("The LAN867X Ethernet PHY requires the ESP-IDF framework.")
+
+        # Check the ESP-IDF version
+        framework_version = CORE.data[KEY_CORE][KEY_FRAMEWORK_VERSION]
+        required_version = cv.Version(5, 3, 1)
+        if framework_version < required_version:
+            raise cv.Invalid(
+                f"The LAN867X Ethernet PHY is only supported on ESP-IDF {required_version} and later."
+            )
```

Comment:
Possibly this can be replaced by the below (untested)
```suggestion
        cv.require_framework_version(esp_idf=cv.Version(5, 3, 1))
```

Reply:
```
validator = cv.require_framework_version(esp_idf=cv.Version(5, 3, 0))
validator(config)
```
works